kernel/include/ksched.h

```diff
@@ -12,13 +12,16 @@
 #include <tracing.h>
 #include <stdbool.h>
 
+BUILD_ASSERT(K_LOWEST_APPLICATION_THREAD_PRIO
+	     >= K_HIGHEST_APPLICATION_THREAD_PRIO);
+
 #ifdef CONFIG_MULTITHREADING
-#define Z_VALID_PRIO(prio, entry_point) \
+#define Z_VALID_PRIO(prio, entry_point)				     \
 	(((prio) == K_IDLE_PRIO && z_is_idle_thread(entry_point)) || \
-		 (z_is_prio_higher_or_equal((prio), \
-			K_LOWEST_APPLICATION_THREAD_PRIO) && \
-		  z_is_prio_lower_or_equal((prio), \
-			K_HIGHEST_APPLICATION_THREAD_PRIO)))
+	 ((K_LOWEST_APPLICATION_THREAD_PRIO			     \
+	   >= K_HIGHEST_APPLICATION_THREAD_PRIO)		     \
```

Comment:
Isn't this already checked in the build assertion you added?  The optimizer will discard the test so there's no impact, except that I think it clutters the macro.

Reply:
The problem is that the macro as written is confusing coverity, see the linked bugs.  Indeed this is a noop, but it allows a static analyzer to prove the expression is correctly in bounds locally without having to inline a bunch of priority predicates and reason about preprocessor values.

(I'm like 40% confident this is actually a coverity bug, but I'm too cautious to just turn off an assertion immediately before a release.  Later on we can come by and rip this out entirely)

Reply:
To be clearer about the logic: I added that test to the macro first to fix the static analysis glitching.  But that would have produced an incorrect "valid" check in the case where the bounds were set wrong, so I check those separately in the build assert to avoid opening a hole. :)